docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,251 @@ Note that:
+
+  3) C8Y config plugin: notifies the external device `child1` via MQTT to upload its current configuration to the thin-edge device
+     * Topic:   `tedge/<childid>/commands/req/config_snapshot/<config file path>`
+     * Example: `tedge/child1/commands/req/config_snapshot/path/to/file/foo.conf`
+     * Payload: `<file-id>`
+     * See also [filetransfer specification](https://github.com/thin-edge/thin-edge.io_spikes/blob/baf5c4f7082ae9cc64b9cd41e9366fac1ae3e2c6/file-transfer-service/SUMMARY.md#configuration-management-mqtt-apis) 
+
+  4) external device child1: notifies the plugin about the "execution" status of the upload operations.
+
+  5) external device child1: Uploads configuration file to the thin-edge device with the HTTP file transfer feature.
+     * HTTP PUT request: `http://<ip address of thin-edge devicee>/tedge/tmpfiles/<file-id>`
+    
+  6) external device child1: notifies the plugin via MQTT about succeeded upload.
+     * Topic:   `tedge/<childid>/commands/res/successful/config_snapshot/<config file path>`
+     * Example: `tedge/child1/commands/res/successful/config_snapshot/path/to/file/foo.conf`
+     * Payload: `<EMPTY>`
+     * See also [filetransfer specification](https://github.com/thin-edge/thin-edge.io_spikes/blob/baf5c4f7082ae9cc64b9cd41e9366fac1ae3e2c6/file-transfer-service/SUMMARY.md#configuration-management-mqtt-apis) 
+ 
+  7) C8Y config plugin: recognizes the MQTT notification about the uploaded file, 
+                       and downloads the file using HTTP file transfer with the `file-id` to some temporary location.
+     * Example for HTTP GET request: `http://<ip address of thin-edge device>/tmpfiles/<file-id>`
+
+  8) C8Y config plugin: sends a delete request to the HTTP file transfer with the `file-id`
+     * Example for HTTP DELETE request: `http://<ip address of thin-edge device>/tmpfiles/<file-id>`
+
+  9) C8Y config plugin: sends the downloaded file to C8Y
+
+  10) C8Y config plugin: removes the downloaded file from temporary location
+  
+
+## Pushing configuration file update to child device from cloud
+
+A configuration file snapshot is pushed to an external device as illustrated below.
+
+```mermaid
+sequenceDiagram
+    participant External Device
+    participant Tedge Agent
+    participant C8Y Cfg Plugin
+    participant C8Y Cloud
+    
+
+        C8Y Cloud->>C8Y Cfg Plugin: 1: ConfigDownloadRequest (type="foo.conf", childid="child1", url="some.url")
+        C8Y Cfg Plugin->>C8Y Cloud: 2: Download new config-file (url)
+
+        C8Y Cfg Plugin->>C8Y Cfg Plugin: 3: Generate random "file-id"     
+        C8Y Cfg Plugin->>Tedge Agent: 4: HTTP PUT /tedge/tmpfiles/<file-id> (content of new config-file)
+
+        C8Y Cfg Plugin->>External Device: 5: MQTT notification for "child1": Requst to download config file "/path/to/file/foo.conf" on url "file-id"
+
+        External Device->>C8Y Cfg Plugin: 6: MQTT notification: Operation Status upload config "Executing" 
```

Comment:
Forwarding the `Executing` status update to cloud is missing here.

Reply:
was now added